Treewide cleanup meta description

afh opened the pull request:
## Description of changes

* Remove ending period from `meta.description` as recommended in [section **Meta attributes** in `pkgs.README.md`](https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes)
* Add recommendation to avoid `meta.description` to start with the definite or an indefinite article to `pkgs/README.md`
* Remove the definite or an indefinite article from the beginning of `meta.description`

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

afh committed:
treewide: Remove ending period from meta.description

nix run nixpkgs#silver-searcher -- -G '\.nix$' -0l 'description.*".*\.";' pkgs \
  | xargs -0 nix run nixpkgs#gnused -- -i '' -Ee 's/(description.*)\.";/\1";/'

afh committed:
treewide: Remove indefinite article from meta.description

nix run nixpkgs#silver-searcher -- -G '\.nix$' -0l 'description.*"[Aa]n?' pkgs \
  | xargs -0 nix run nixpkgs#gnused -- -i '' -Ee 's/(description.*")[Aa]n? (.)/\1\U\2/'

afh committed:
treewide: Remove the definite article from meta.description

nix run nixpkgs#silver-searcher -- -G '\.nix$' -0l 'description.*"([Tt]he)? ' pkgs \
  | xargs -0 nix run nixpkgs#gnused -- -i '' -Ee 's/(description.*")[Tt]he (.)/\1\U\2/'

jtojnar commented on pkgs/applications/audio/aucatctl/default.nix:
```diff
@@ -27,7 +27,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   meta = with lib; {
-    description = "The aucatctl utility sends MIDI messages to control sndiod and/or aucat volumes";
+    description = "Aucatctl utility sends MIDI messages to control sndiod and/or aucat volumes";
```
Name of the utility probably should not be uppercased.

afh replied:
Thanks @jtojnar for chiming in on this, much appreciated.

According to the recommendation in [`pkgs.README`](https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes) the description should "[n]ot start with the package name", so it might be appropriate to remove it altogether. What do you think? I'll try to come up with a scripted way to address this recommendation too.


```suggestion
    description = "Utility sends MIDI messages to control sndiod and/or aucat volumes";
```

mweinelt replied:
In this case having the program name as the start looks good to me. Replacing it with just "Utility" makes no sense.

mweinelt replied:
> Utitility that allows sending MIDI Messages....

afh replied:
Any preference in terms of capitalisation, @mweinelt?

sternenseemann commented:
`hackage-packages.nix` is automatically generated from upstream maintainer provided information. Editing this file serves nothing, but create merge conflicts as any changes will be reverted. This is clearly stated at the beginning of the file.

afh commented:
Thanks for calling this out, @sternenseemann, very much appreciated! While working on the changes I focused on the created diffs and did not think about generated nix files.

I've reworked this PR to omit changes to the following generated files:
* pkgs/development/haskell-modules/hackage-packages.nix
* pkgs/development/compilers/elm/packages/node/node-packages.nix
* pkgs/development/lua-modules/generated-packages.nix
* pkgs/development/node-packages/node-packages.nix
* pkgs/misc/base16-builder/node-packages-generated.nix

afh commented:
✅ All checks have passed 🎉 

Though I did not expect as many rebuilds with just changes to `meta.description`. Will folks accept this PR to `master` regardless or should the merge base be changed to `staging`?

K900 commented:
I personally would very much like to know what rebuilds this is causing and why.

afh commented:
Couldn't agree more, @K900! 💯 

The commands that were used targeted _any_ files that matched the regexes, instead of only targeted files ending in `\.nix` (for details see the commit messages). That included patches and json files, the latter possibly being used for dependencies, both of which could explain the rebuilds.
I went ahead and removed the changes to non-nix files from this PR. Let's see what ofborg has to say about this.

❓ Will ofborg update the labels accordingly or has this PR now been tainted with the mass-rebuild tags?

K900 commented:
It will update when it's done evaluating.

wolfgangwalther reviewed (commented):
I went through everything in `pkgs/applications/audio` and `pkgs/applications/backup`.

Capitalizing the first word after the removed article doesn't always work out best.

I looked at 214 files, having 5 comments, that's about 2%.

afh commented on pkgs/applications/audio/freac/default.nix:
```diff
@@ -29,7 +29,7 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
-    description = "The fre:ac audio converter project";
+    description = "Fre:ac audio converter project";
```
How about the following extracted from https://www.freac.org?
```suggestion
    description = "Audio converter and CD ripper with support for various popular formats and encoders.";
```

afh replied:
You saw nothing! 😅

K900 reviewed (dismissed):
Just a general -1 until this is 0 rebuilds.

afh commented on pkgs/applications/audio/moc/default.nix:
```diff
@@ -91,7 +91,7 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
-    description = "An ncurses console audio player designed to be powerful and easy to use";
+    description = "Ncurses console audio player designed to be powerful and easy to use";
```
What are your thoughts on this:
```suggestion
    description = "Console audio player designed to be powerful and easy to use";
```

afh commented on pkgs/applications/audio/ncpamixer/default.nix:
```diff
@@ -34,7 +34,7 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
-    description = "An ncurses mixer for PulseAudio inspired by pavucontrol";
+    description = "Ncurses mixer for PulseAudio inspired by pavucontrol";
```
Similar to the previous comment above. Question is what is more is helpful if the description lists technologies, e.g. ncurses.
```suggestion
    description = "Console mixer for PulseAudio inspired by pavucontrol";
```

afh commented:
Thank you for taking the time to go over some of the changes proposed here, @wolfgangwalther, much appreciated. I left a response to some of your comments with a different take on the description and will rework the others.

afh commented:
I hear you, @K900. As I'm digging deeper to find what is causing the rebuilds, do you or anyone else, have ideas where these could come from?

afh committed:
treewide: Improve wording of meta.description

mweinelt commented:
```
abaddon.x86_64-linux                                                                     /nix/store/gddx7cl2cgxm4sp61n85xs2bi3wgapln-abaddon-0.2.1
amphetype.x86_64-linux                                                                   dist=/nix/store/68v75fx5fbg0d9lz5bf2jpf92h4vk92g-amphetype-1.0.0-dist;/nix/store/fb52lyrdwf6k7sjk8w7ks27wziqabbzj-amphetype-1.0.0
appflowy.x86_64-linux                                                                    /nix/store/clsnbp5ydh4yxrwj4k6dnczgzjr5cha4-appflowy-0.5.7
bitwarden-desktop.x86_64-linux                                                           /nix/store/zjadxnd55wxcy2iafcibjmfhnlyzz2g6-bitwarden-desktop-2024.4.1
bitwuzla.x86_64-linux                                                                    /nix/store/q8kmcq2cwx5hjf4613p6aj0kk1yrxr0i-bitwuzla-0.4.0
broken-sword-25.x86_64-linux                                                             /nix/store/n3x55vphib6rfv7f3vkky1q1i6yicj6d-broken-sword-2.5-1.0
camunda-modeler.x86_64-linux                                                             /nix/store/yxak45qnn66kv44b8mrchm19s6az9lxl-camunda-modeler-5.23.0
cvc5.x86_64-linux                                                                        /nix/store/dak75srlzxkalh9grvc9ji9rnr82zclk-cvc5-1.1.2
element-desktop-wayland.x86_64-linux                                                     /nix/store/a5az7bn1barrx43fgaix8ggd4via4xc2-element-desktop
element-desktop.x86_64-linux                                                             /nix/store/cs3nppj2w67w46k3j1b2czp95pmwcf5l-element-desktop-1.11.67
epgstation.x86_64-linux                                                                  /nix/store/nahqwad7qdak3qm98m0pk54gzi6vkybk-epgstation-2.6.20
experienced-pixel-dungeon.x86_64-linux                                                   /nix/store/kdviyncfx7fp3fn7jmg4k16aq88f1n1r-experienced-pixel-dungeon-2.18
fastqc.x86_64-linux                                                                      /nix/store/2xxj251f3ha33cqqa0k627zq82pb62h9-fastqc-0.12.1
gitnuro.x86_64-linux                                                                     /nix/store/ac7n80q88d4jilq1i5240rndmi01nvjy-gitnuro-1.3.1
gomuks.x86_64-linux                                                                      /nix/store/p872j8iyzzbikknsb613dkpps8l21n4m-gomuks-0.3.0
hase.x86_64-linux                                                                        /nix/store/mzx1jcqcxy8lh8l1f82j72z24f64z1zq-hase-unstable-2020-10-06
hdfview.x86_64-linux                                                                     /nix/store/zxfc9684kz3h82bkqm45j263617zpby4-hdfview-3.3.1
hmcl.x86_64-linux                                                                        /nix/store/qijmhsqainkj7vd5j9sj7g6m9fqk2w76-hmcl-3.5.8
hyperrogue.x86_64-linux                                                                  /nix/store/aqk4qrxicwba5l2pdmxk9zwy5gzdwana-hyperrogue-13.0i
inochi-creator.x86_64-linux                                                              /nix/store/qa76afj5fpm20m2c24b33zzdw6588yq6-inochi-creator-0.8.5
inochi-session.x86_64-linux                                                              /nix/store/c85mgnzd51z26jixac7gkynw127xyzrf-inochi-session-0.8.4
isabelle-components.isabelle-linter.x86_64-linux                                         /nix/store/ka1w5bz0bii0bxszv8yd7xljqyqxi606-isabelle-linter-2023-1.0.0
isabelle.x86_64-linux                                                                    /nix/store/1syg1qla27jvms26rsfkp2xr7z6fcvmm-isabelle-2023
koodo-reader.x86_64-linux                                                                /nix/store/dk2358j9igcxscwml139jzfnmc22scnn-koodo-reader-1.6.6
kuro.x86_64-linux                                                                        /nix/store/nvrri0z6hfasnlm36p0c7pdnchi0p419-kuro-9.0.0
leo-editor.x86_64-linux                                                                  /nix/store/wp8x32izsrvmqgm3fgjlw51d4gr4arsd-leo-editor-6.7.8
liana.x86_64-linux                                                                       /nix/store/wbbm1wxzg6833fahxgsp3fbvqsl71wan-liana-5.0
lorien.x86_64-linux                                                                      /nix/store/fqr1fgmpzdln2cmynkrayfqpx4xzbpb8-lorien-0.6.0
lumafly.x86_64-linux                                                                     /nix/store/4cm9yymicykdwlnr826n8qnkmjmz1lkl-lumafly-3.2.0.0
marble-marcher-ce.x86_64-linux                                                           /nix/store/xpdl713fziljchrk3x46v7yspwrk8f84-marble-marcher-ce-1.4.5
mars-mips.x86_64-linux                                                                   /nix/store/fzs01d0mw2s4ljcjn7a5mckh8b3rrpxy-mars-mips-4.5
micropad.x86_64-linux                                                                    /nix/store/jflxgh7g4i82sksv7vlva9czvbk0vvyf-micropad-4.5.1
mikutter.x86_64-linux                                                                    /nix/store/a10yc86sh5cs8czm86mavb1q6d6fh3d9-mikutter-4.1.4
mirakurun.x86_64-linux                                                                   /nix/store/d76p4383dxf4g4slckx2c3z0wbdg0x4v-mirakurun-3.9.0-rc.4
openraPackages_2019.mods.ca.x86_64-linux                                                 /nix/store/4ncrsaavxfy4hqsx7g4kqq2nr523b48b-openra_2019-ca-96.git.fc3cf0b
openraPackages_2019.mods.d2.x86_64-linux                                                 /nix/store/lc0gmi01m257lv935nnzc88y07i1mwh0-openra_2019-d2-134.git.69a4aa7
openraPackages_2019.mods.dr.x86_64-linux                                                 /nix/store/wr5xxq9897bi2dw9g2k11pzpdyr8571k-openra_2019-dr-324.git.ffcd6ba
openraPackages_2019.mods.mw.x86_64-linux                                                 /nix/store/izsg0szly1874qh2pczmv3vrqzcq08ir-openra_2019-mw-257.git.c9be8f2
openraPackages_2019.mods.raclassic.x86_64-linux                                          /nix/store/bb09kxzkxvnnlw29g8qanzvmimzcrr3q-openra_2019-raclassic-183.git.c76c13e
openraPackages_2019.mods.ss.x86_64-linux                                                 /nix/store/bzz5m1snailkg0zjbw4jr6401iz7dmjn-openra_2019-ss-77.git.23e1f3e
opensupaplex.x86_64-linux                                                                /nix/store/cf8hjc0zs1a3w2pgfax11gjnzlrzsk57-opensupaplex-7.1.2
podman-desktop.x86_64-linux                                                              /nix/store/cagmf32x4cm8km4c26wg8574hwd0w077-podman-desktop-0.12.0
popcorntime.x86_64-linux                                                                 /nix/store/6w9jc9jdk0lw43h5cz6vlsbjm8cq7wfp-popcorntime-0.5.1
premid.x86_64-linux                                                                      /nix/store/10nb23bkzz6bgpwrpdhjbw3x161dffxq-premid-2.3.4
protege.x86_64-linux                                                                     /nix/store/mfmc465glz2db4hqbpimlwchc6g71mbn-protege-5.6.4
pymol.x86_64-linux                                                                       dist=/nix/store/32k66wxlzb9fv1nwr0if5r2dw48arhv4-pymol-3.0.0-dist;/nix/store/94gw08zd5wgaryihhhmxxmznkwwlnbj3-pymol-3.0.0
pysolfc.x86_64-linux                                                                     dist=/nix/store/b4xp5hv9f1qhja3xd96w9sgyfbc8a15z-pysolfc-2.21.0-dist;/nix/store/zjzbrx11zrybfvfn2krf4falb6saypfj-pysolfc-2.21.0
python311Packages.raylib-python-cffi.x86_64-linux                                        dist=/nix/store/pij9kq3yclnk3q0zdd3hlv7f6n4sbz3n-python3.11-raylib-python-cffi-5.0.0.2-dist;/nix/store/h85d1sxbmn314w1fyy3nwjar9a38ddkh-python3.11-raylib-python-cffi-5.0.0.2
python312Packages.raylib-python-cffi.x86_64-linux                                        dist=/nix/store/als939jh3893wii2ws45l1x8qg1b0l34-python3.12-raylib-python-cffi-5.0.0.2-dist;/nix/store/3cf6f21bhrxqv6b0vk4m8pq64gh039z1-python3.12-raylib-python-cffi-5.0.0.2
quark-goldleaf.x86_64-linux                                                              /nix/store/vq08fc9vrl7nszbk01cpji7vv4rx8q63-quark-goldleaf-1.0.0
rat-king-adventure.x86_64-linux                                                          /nix/store/lr3ikj90g9fffz1havl27avkjisr0g5z-rat-king-adventure-1.5.3
raygui.x86_64-linux                                                                      /nix/store/7x9bjbkmb6j2ag4l3nhjqxps6lx0pk8z-raygui
release-checks                                                                           /nix/store/ankdrb564xzvmm6aigdkfv054ai57cxg-nixpkgs-release-checks
rxvt-unicode-emoji.x86_64-linux                                                          /nix/store/5d9qhp9wilj4xsi0d92xywpxnyqx09n8-rxvt-unicode-9.31
rxvt-unicode-unwrapped-emoji.x86_64-linux                                                /nix/store/lqr294pfzzd8yhdsn55vsd79z1d0p3qc-rxvt-unicode-unwrapped-9.31;terminfo=/nix/store/205zpwla334njl0x0llsaiwn7d9w3vzk-rxvt-unicode-unwrapped-9.31-terminfo
rxvt-unicode-unwrapped.x86_64-linux                                                      /nix/store/f8363zwcd54fn0fsgywcmah508ih8rx7-rxvt-unicode-unwrapped-9.31;terminfo=/nix/store/dij4zp9m950rsamipkmxnrklmkl37i2l-rxvt-unicode-unwrapped-9.31-terminfo
rxvt-unicode.x86_64-linux                                                                /nix/store/xc3b8bysapdnwr2xk5ccjsbbk4dhrqvq-rxvt-unicode-9.31
shorter-pixel-dungeon.x86_64-linux                                                       /nix/store/cc4h2b8mrgdxj7pkn22375p6wym900g4-shorter-pixel-dungeon-1.4.0
sparrow3d.x86_64-linux                                                                   dev=/nix/store/skrasddkgpp1ciamqgsbgwsqblf20f07-sparrow3d-unstable-2020-10-06-dev;/nix/store/kxwqr8smh8ddl5whyqp0vl5ggfpqm8dy-sparrow3d-unstable-2020-10-06
subtitleedit.x86_64-linux                                                                /nix/store/r2f5q60p3y7r5g247a1kr0wkzaqglpxp-subtitleedit-4.0.5
summoning-pixel-dungeon.x86_64-linux                                                     /nix/store/a11x0wcvk4kmxrw1506y9q17i9zmq0yz-summoning-pixel-dungeon-1.2.5a
supersonic-wayland.x86_64-linux                                                          /nix/store/113l0f90blsp62p931rqf0hamsi3prah-supersonic-wayland-0.10.1
supersonic.x86_64-linux                                                                  /nix/store/sc2mppyrr4676d5hwap9pcl6qb770y16-supersonic-0.10.1
symfpu.x86_64-linux                                                                      /nix/store/s7gdmkqgmmrchk1pdcf0d37a30qhsqsq-symfpu-unstable-2019-05-17
techmino.x86_64-linux                                                                    /nix/store/f6p05bw8sgabc1pcc72hnsy6is6ffxav-techmino-0.17.16
tlaps.x86_64-linux                                                                       /nix/store/2wh5scpqmrwdmc9q25fd0sg0b8pfw109-tlaps-1.4.5
transgui.x86_64-linux                                                                    /nix/store/dd22jzb8zfzngjjwcrqcwsvbp6xmimyx-transgui-5.18.0-unstable-2024-02-26
vis.x86_64-linux                                                                         /nix/store/qsnsl0axn09nlxp8dfx87aisc6jw0g50-vis-0.9
webcord-vencord.x86_64-linux                                                             /nix/store/gbjg9cwjmb6xacj8xmh9993qs9rvp0l4-webcord-vencord-4.9.2
webcord.x86_64-linux                                                                     /nix/store/96b8x4vr9f8zq3bdcpd16a6rcdrxq9h2-webcord-4.9.2

afh commented:
Thanks for looking into this, @mweinelt, and sharing the list of `x86_64-linux` rebuilds. Which command(s) did you use to generate the list?

I thought that the rebuilds might've come from the changes to `meta.description` in `pkgs/stdenv/generic/default.nix`, but reverting that change unfortunately did not decrease the number of rebuilds.

mweinelt commented:
> Which command(s) did you use to generate the list?

```
maintainer/scripts/rebuild-amount.sh --print
```

afh commented:
Thanks, @wolfgangwalther, that's very helpful. I'll look into this and make the appropriate changes to this PR.

That's good to know, @mweinelt, thanks! Running `./maintainers/scripts/rebuild-amount.sh --print master | head -6` on this PR branch in a pristine git worktree reports:

```
Estimating rebuild amount by counting changed Hydra jobs (parallel=unset).
   1 pkgs-lib-tests
   1 release-checks
49808 x86_64-darwin
72937 x86_64-linux
```

this differs by orders of magnitude from ofborg's and your report. 
What am I doing wrong?

mweinelt commented:
The base branch has moved on. Try running it with `HEAD~4` instead.

afh commented:
I'm not sure that's really it as the `master` branch in my local repo hasn't moved on:

```sh
git log --oneline master..HEAD
  d42bb3333f78 (HEAD -> treewide-cleanup-meta-description, origin/treewide-cleanup-meta-descriptio  n) treewide: Improve wording of meta.description
  ce8e26495cf1 treewide: Remove the definite article from meta.description
  84b5eb1d233e treewide: Remove indefinite article from meta.description
  c4182ec375eb treewide: Remove ending period from meta.description
```

When running `./maintainers/scripts/rebuild-amount.sh --print HEAD~4` the following error is printed:
```sh
error: syntax error, unexpected PATH_END, expecting DOLLAR_CURLY

       at /var/folders/h0/948x8bgn0kd7b1d7hf0bnlr00000gp/T/nix-rebuild-amount-FKM2cs8b:2:73:

            1|         let
            2|           lib = import /var/folders/h0/948x8bgn0kd7b1d7hf0bnlr00000gp/T//nix-rebuild-amount-r1Ymbr2y/lib;
````

afh commented:
@wolfgangwalther Another pattern I found is a `comment = description` with a `let … in` block. I went ahead and updated, i.e. removed those changes to `description`, the commits in this PR; let's see what ofborg says about this.

Until I'm able to reliably and "correctly" run `./maintainers/scripts/rebuild-amount.sh` would you, @mweinelt, or someone else might be so kind to run it on my behalf to see which rebuilds remain?

mweinelt commented:
```
      1 release-checks
      3 x86_64-darwin
     17 x86_64-linux

bitwuzla.x86_64-darwin                                                                   /nix/store/30m205wsdy1naacfffv169yr54khqq7a-bitwuzla-0.5.0
bitwuzla.x86_64-linux                                                                    /nix/store/rkln14w6dkdrn1sk43xczfvaaydvpg99-bitwuzla-0.5.0
cvc5.x86_64-darwin                                                                       /nix/store/45pnm5cjq783ylw7mbzpnja79q7dwhgb-cvc5-1.1.2
cvc5.x86_64-linux                                                                        /nix/store/dak75srlzxkalh9grvc9ji9rnr82zclk-cvc5-1.1.2
experienced-pixel-dungeon.x86_64-linux                                                   /nix/store/kdviyncfx7fp3fn7jmg4k16aq88f1n1r-experienced-pixel-dungeon-2.18
hase.x86_64-linux                                                                        /nix/store/mzx1jcqcxy8lh8l1f82j72z24f64z1zq-hase-unstable-2020-10-06
inochi-creator.x86_64-linux                                                              /nix/store/8mk28pbc22f2wi5m00razrvc9pxyhh56-inochi-creator-0.8.5
inochi-session.x86_64-linux                                                              /nix/store/vh075f3v2m9b24f8l4hrw5xns99dq164-inochi-session-0.8.4
openraPackages_2019.mods.ca.x86_64-linux                                                 /nix/store/4ncrsaavxfy4hqsx7g4kqq2nr523b48b-openra_2019-ca-96.git.fc3cf0b
openraPackages_2019.mods.d2.x86_64-linux                                                 /nix/store/lc0gmi01m257lv935nnzc88y07i1mwh0-openra_2019-d2-134.git.69a4aa7
openraPackages_2019.mods.dr.x86_64-linux                                                 /nix/store/wr5xxq9897bi2dw9g2k11pzpdyr8571k-openra_2019-dr-324.git.ffcd6ba
openraPackages_2019.mods.mw.x86_64-linux                                                 /nix/store/izsg0szly1874qh2pczmv3vrqzcq08ir-openra_2019-mw-257.git.c9be8f2
openraPackages_2019.mods.raclassic.x86_64-linux                                          /nix/store/bb09kxzkxvnnlw29g8qanzvmimzcrr3q-openra_2019-raclassic-183.git.c76c13e
openraPackages_2019.mods.ss.x86_64-linux                                                 /nix/store/bzz5m1snailkg0zjbw4jr6401iz7dmjn-openra_2019-ss-77.git.23e1f3e
rat-king-adventure.x86_64-linux                                                          /nix/store/lr3ikj90g9fffz1havl27avkjisr0g5z-rat-king-adventure-1.5.3
release-checks                                                                           /nix/store/0r800i0f1mmhr48444620p3yzcdpc2i9-nixpkgs-release-checks
shorter-pixel-dungeon.x86_64-linux                                                       /nix/store/cc4h2b8mrgdxj7pkn22375p6wym900g4-shorter-pixel-dungeon-1.4.0
sparrow3d.x86_64-linux                                                                   dev=/nix/store/skrasddkgpp1ciamqgsbgwsqblf20f07-sparrow3d-unstable-2020-10-06-dev;/nix/store/kxwqr8smh8ddl5whyqp0vl5ggfpqm8dy-sparrow3d-unstable-2020-10-06
summoning-pixel-dungeon.x86_64-linux                                                     /nix/store/a11x0wcvk4kmxrw1506y9q17i9zmq0yz-summoning-pixel-dungeon-1.2.5a
symfpu.x86_64-darwin                                                                     /nix/store/mrff1pcxyc8gvdzghx3mqy3ym3mqwrml-symfpu-unstable-2019-05-17
symfpu.x86_64-linux                                                                      /nix/store/s7gdmkqgmmrchk1pdcf0d37a30qhsqsq-symfpu-unstable-2019-05-17

afh commented:
Very helpful, @mweinelt, thanks a bunch! I addressed the issues I found (see details below), mind generating the list again?

* bitwuzla and cvc5 have a dependency on symfpu, so if symfpu is built those two packages also get build
* hase has a dependency sparrow3d
* symfpu  and sparrow3d `inherit` `dependency` in `pkgconfigItems`
* inochi2d, rat-king-adventure, summoning-pixel-dungeon, shorter-pixel-dungeon, experienced-pixel-dungeon uses `comment = description` in `generic.nix` from which the packages are built
* openraPackages_2019 uses description installPhase with substitute, which likely triggers a rebuild.

mweinelt commented:
```
      1 release-checks

release-checks                                                                           /nix/store/5vmlq9q78vp9xlsvf58ncb4p82274wpx-nixpkgs-release-checks

afh commented:
Thanks once more, @mweinelt, very much appreciated! :pray:

I'm looking at `pkgs/top-level/nixpkgs-basic-release-checks.nix` which seems to be what is run as `release-checks`, but I'm not familiar with those aspects of nixpkgs. Can someone more knowledgable chime in on what changes in this PR could cause `release-checks` to be rebuilt?

@K900 would you be open to re-consider and potentially give your approval to the changes proposed here with 1 rebuild?

K900 commented:
There's merge conflicts now.

afh commented:
The merge conflicts have been addressed, @K900. Given the amount of changes in this PR and the many contributions to this project it seems there is narrower merge window. I'll do my best too keep this up-to-date and address merge conflicts.

afh commented:
Thanks everyone for lending your expertise on this PR and helping getting it merged, very much appreciated! 😃🙏

Aleksanaa commented:
> Add recommendation to avoid meta.description to start with the definite or an indefinite article to pkgs/README.md

Is there any reason for us to do this? Is there any prior context for adding this rule here?

IMO starting the description with "A", "An" or "The" does not affect the correctness of the description and makes it read more naturally. I don't understand at all why we need to avoid it.

Aleksanaa commented:
Let me give you an example: "An implementation of" and "implementation of" can be interpreted to different meanings. The latter seems even less clear: is this an official/standard/only implementation, or just an implementation?

Aleksanaa commented:
In addition, you should not modify the document and perform treewide changes in one commit. This makes the situation even more unclear to reviewers: although you gave the command, since I found some changes that were obviously not completed by this command, So how do I ensure that all the modifications in your commit (which contains thousands of files and can even freeze Firefox when browsing online) are reliable?

The modification of the document is also incomplete. You have not modified the corresponding part in the nixpkgs manual, which creates more confusion. The manual still describes it as "right": https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#description-var-meta-description

afh commented:
Thank you, @Aleksanaa, for chiming in with helpful feedback, very much appreciated, :pray:

> Is there any reason for us to do this?

To keep the description succinct and concise and free from words that provide no additional meaning.

> Is there any prior context for adding this rule here?

[OpenBSD's Porting Guide (see 18.)](https://www.openbsd.org/faq/ports/guide.html) shares similar recommendations as the the nixpkgs' `pkgs/README.md` and recommends to **_"remove the article altogether:"_**

> _Add COMMENT in Makefile. COMMENT is a SHORT one-line description of the port (max. 60 characters). Do NOT include the package name (or version number of the software) in the comment. Do NOT start with an uppercase letter unless semantically significant, and do NOT end with a period. DON'T EVER START WITH AN INDEFINITE ARTICLE SUCH AS "a" or "an" - **remove the article altogether.**_ 

> The latter seems even less clear: is this an official/standard/only implementation, or just an implementation?

If it is an official or standard implementation the description should include that adjective, if no such adjective is present one can safely assume it is just an implementation.

> you should not modify the document and perform treewide changes in one commit.

So one commit for the doc change and one for the treewide application?

> I found some changes that were obviously not completed by this command

Which examples did you find?

> how do I ensure that all the modifications in your commit (which contains thousands of files and can even freeze Firefox when browsing online) are reliable

Unfortunately I do not have a good answer for that from the top of my head. Let me give it some thought.

> You have not modified the corresponding part in the nixpkgs manual

Apologies for that I thought the `pkgs/README.md` was the authoritative source and I'll gladly create a follow-up PR to remedy the shortcoming(s) of this PR.

Hopefully I've addressed all your comments and concerns, in case I something left out, please let me know.

Aleksanaa commented:
> To keep the description succinct and concise and free from words that provide no additional meaning.

IMO The smooth and less ambiguous grammar should be the first guarantee. Deleting "A/An/THE" does not significantly reduce the time of reading or reduce the size of the repository.

> [OpenBSD's Porting Guide (see 18.)](https://www.openbsd.org/faq/ports/guide.html) shares similar recommendations as the the nixpkgs' pkgs/README.md and recommends to "remove the article altogether:"

Okay, that's at least the same rule from somewhere. I just checked and it seems that Debian does not add the A/An/The prefix, while Arch and Fedora do not restrict this. Of course I'm not saying that OpenBSD is the "model distribution", or that Fedora and Arch are. But attitudes on this issue are clearly divided.

I think this needs further discussion. This may have something to do with native language habits. Some other people in the Chinese group also complained to me that removing the prefix Definite/Indefinite article makes no sense. However, I don't know why, given a controversial decision, this PR was merged within a few days without any discussion about it. This is obviously inappropriate.

> Which examples did you find?

I mean editing the doc in one commit with the treewide change.

> Unfortunately I do not have a good answer for that from the top of my head. Let me give it some thought.

If you just give a command (thanks for giving the command at least), we can directly reproduce the diff. If you throw some other modifications into it (I mean the documentation, other parts haven't been fully checked yet), things get a little more difficult.

eclairevoyant commented:
> Let me give you an example: "An implementation of" and "implementation of" can be interpreted to different meanings. The latter seems even less clear: is this an official/standard/only implementation, or just an implementation?

"The implementation" does not imply standard impl either. If it's really to imply that it's the standard impl, then put the word "standard" or "reference" in the desc and be completely unambiguous. And to call something "the implementation" is misleading; it might be the only impl at the time... until it's not. I find it very difficult to believe that removing leading articles would impede understanding.
